geode-core/src/main/java/org/apache/geode/distributed/internal/membership/InternalDistributedMember.java

```diff
@@ -637,7 +638,68 @@ public boolean equals(Object obj) {
     if ((obj == null) || !(obj instanceof InternalDistributedMember)) {
       return false;
     }
-    return compareTo((InternalDistributedMember) obj) == 0;
+    InternalDistributedMember other = (InternalDistributedMember) obj;
+
+    int myPort = getPort();
+    int otherPort = other.getPort();
+    if (myPort != otherPort) {
+      return false;
+    }
+
+    InetAddress myAddr = getInetAddress();
```

Comment:
Could we replace may of these blocks of code with.
```
if (!Object.equals(getInetAddress(), other.getInetAddress())) {
  return false;
}

Reply:
That is a good idea. We would not want to use it on primitive fields (like the ports) and the inetAddress comparison has a special case that if they are both null then we return true immediately. I will make the change to use Objects.equals where possible and update this PR.